### ISSUE TITLE
Remap loses `loc` and `decl` properties used by nyc reports

### ISSUE THREAD

**Issue by DanTup** [SitePen/remap-istanbul#163]
My source files collected using `nyc instrument` have json objects with `decl` properties in the `fnMap` and `loc` properties in the `branchMap`. After running through `remap-istanbul` both of these properties are lost and therefore `nyc report` chokes when trying to generate any details reports (html, lcov).

Source:

```js
{
	"project.js": {
		"path": "project.js",
		"fnMap": {
			"0": {
				"name": "locateBestProjectRoot",
				"decl": {
					"start": { "line": 6, "column": 9 },
					"end": { "line": 6, "column": 30 }
				},
				"loc": {
					"start": { "line": 6, "column": 39 },
					"end": { "line": 16, "column": 1 }
				},
				"line": 6
			}
		},
		"branchMap": {
			"0": {
				"loc": {
					"start": { "line": 7, "column": 4 },
					"end": { "line": 8, "column": 20 }
				},
				"type": "if",
				"locations": [
					{
						"start": { "line": 7, "column": 4 },
						"end": { "line": 8, "column": 20 }
					},
					{
						"start": { "line": 7, "column": 4 },
						"end": { "line": 8, "column": 20 }
					}
				],
				"line": 7
			}
		}
	}
}
```

Output:

```js
{
	"../../src/project.ts": {
		"path": "../../src/project.ts",
		"fnMap": {
			"1": {
				"name": "locateBestProjectRoot",
				"line": 6,
				"loc": {
					"start": { "line": 6, "column": 52 },
					"end": { "line": 18, "column": 1 }
				}
			}
		},
		"branchMap": {
			"1": {
				"line": 7,
				"type": "if",
				"locations": [
					{
						"start": { "line": 7, "column": 1 },
						"end": { "line": 8, "column": 14 }
					},
					{
						"start": { "line": 7, "column": 1 },
						"end": { "line": 8, "column": 14 }
					}
				]
			}
		}
	}
}
```

**Comment by dylans** [comment]
Hi @DanTup , could you please explain the use case for using nyc with remap-istanbul. I thought that nyc uses the newer version of Istanbul (1.x+) which includes source map support out of the box? Whereas remap-istanbul is primarily used to support pre 1.x versions of Istanbul that did not support source maps in determining code coverage.

**Comment by DanTup** [comment]
Ugh, it's a long story :)

Basically, I have a VS Code extension written in TypeScript. The only way nyc's source maps seem to work for me is if I launch my tests through nyc and it does everything. However, because of how VS Code tests work (you run `code.exe` and it spawns the tests itself) nyc doesn't seem to be able to get its mitts inside. So, I use nyc to instrument the code, then have a `signal-exit` hook inside my tests to dump the coverage to disk.

This comes out unmapped, so I then use remap-istantbul to translate the results.

It all works perfectly except for this bug, and if I rig it to use the location data that's in there, I finally get my reports :)

I think the fix here could be fairly trivial (if there's a `decl`/`loc` just map them and pass them through to the output). I think I can fix it and send a PR though I'm fighting with getting the test working atm.

If you know how I can get nyc to work through being called by an external app, or how I can make nyc do the remapping, I'd love to know. Here's how my scripts look:

https://github.com/Dart-Code/Dart-Code/blob/3ce183cf3918a1b340febf39f3bb90d763d49d04/package.json#L609-L618

And my test script that runs the remapping:

https://github.com/Dart-Code/Dart-Code/blob/3ce183cf3918a1b340febf39f3bb90d763d49d04/test/test_all.ts#L39-L53

I found I had to run remap-istanbul from a specific folder to make it work (the `-b` switch didn't work for me) and that it could only do a file at a time, which is why I added it inside my test script (I run my tests multiple times - for dev/stable versions of Dart, Flutter, Code).
